xml_converter/generators/code_generator.py

```diff
@@ -455,14 +474,50 @@ def generate_cpp_variable_data(
             if doc_type in fieldval['applies_to']:
                 xml_fields = []
                 default_xml_fields = []
-                xml_export = ""
+                xml_child_export = []
+                xml_parent_export = ""
+                is_xml_export = True
+
+                # Compound Values are unique in that the components have xml fields in addition to the compound variable
+                if fieldval['type'] in ["CompoundValue", "CompoundCustomClass"]:
+                    xml_child_export = fieldval['xml_child_export']
+                    xml_parent_export = fieldval['xml_parent_export']
+                    for component in fieldval['components']:
+                        component_xml_fields = []
+                        component_default_xml_fields = []
+                        if component['name'] in fieldval['xml_child_export']:
+                            component_default_xml_fields.append(component['xml_fields'][0])
+                            class_name = documentation_type_data[component['type']]["class_name"]
+                            is_xml_export = True
+                        if component['name'] in fieldval['xml_parent_export']:
```

Comment:
This needs to change, you are searching for substrings in your format string, and also just throwing away the rest of the formatting. This is a very very fuzzy check  with no benefits over having `xml_parent_export` just be an array because you are just making it an array here.